sklearn/preprocessing/tests/test_common.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+import numpy as np
+
+
+from sklearn.datasets import load_iris
+
+from sklearn.model_selection import train_test_split
+
+from sklearn.preprocessing import QuantileTransformer
+
+from sklearn.utils.testing import assert_array_equal
+
+iris = load_iris()
+
+TRANSFORMER_HANDLING_NAN = ['QuantileTransformer']
+
+
+def _generate_tuple_transformer_missing_value():
+    trans_handling_nan = [globals()[trans_name]()
+                          for trans_name in TRANSFORMER_HANDLING_NAN]
+    return [(trans, iris.data.copy(), 15)
+            for trans in trans_handling_nan]
+
+
+@pytest.mark.parametrize(
+    "est, X, n_missing",
+    _generate_tuple_transformer_missing_value()
```

Comment:
I don't get why this is better than just parameterizing est directly

Reply:
We could even consider a list of all feature-wise preprocessors then xfail some...

Reply:
> We could even consider a list of all feature-wise preprocessors then xfail some...

I agree. We could switch to this behaviour when a majority of those preprocessor support this feature.